torch/_meta_registrations.py

```diff
@@ -2076,22 +2077,86 @@ def meta__scaled_dot_product_flash(
         dtype=torch.float,
         device=query.device,
     )
-    is_sm80 = torch.cuda.is_available() and torch.cuda.get_device_capability() >= (8, 0)
-    is_sm75 = torch.cuda.is_available() and torch.cuda.get_device_capability() >= (7, 5)
-    head_size_rounded = 64 if head_dim <= 64 else 128
-    blocksize_c = (
-        128
-        if (head_size_rounded == 128 and (dropout_p != 0.0 or not is_sm80))
-        or (is_sm75 and head_size_rounded == 64 and dropout_p != 0.0)
-        else 256
-    )
-    max_seqlen_k = math.ceil(max_seqlen_batch_k / blocksize_c) * blocksize_c
-    if max_seqlen_k <= 128:
-        max_seqlen_k = 128
-    elif max_seqlen_k <= 256:
-        max_seqlen_k = 256
-
-    return ouput, logsumexp
+    cumulative_sequence_length_q = torch.empty(
+        batch_size + 1, dtype=torch.int32, device="meta"
+    )
+    cumulative_sequence_length_k = torch.empty(
+        batch_size + 1, dtype=torch.int32, device="meta"
+    )
+
+    if return_debug_mask:
+        blocksize_c = 128 if head_dim > 64 else 256
+        max_seqlen_k = math.ceil(max_seqlen_batch_q / blocksize_c)
+        if max_seqlen_batch_k <= 128:
+            max_seqlen_k = 128
+        elif max_seqlen_batch_k <= 256:
+            max_seqlen_k = 256
+        debug_mask = torch.empty(
+            (batch_size, num_heads, max_seqlen_q, max_seqlen_k),
+            dtype=query.dtype,
+            device=query.device,
+        )
+    else:
+        debug_mask = torch.empty(0, dtype=query.dtype, device=query.device)
+
+    return (
+        ouput,
+        logsumexp,
+        cumulative_sequence_length_q,
+        cumulative_sequence_length_k,
+        max_seqlen_batch_q,
+        max_seqlen_batch_k,
+        1,  # what do I put here
```

Comment:
This does not work.

I have a minimal repro script that compiles SDPA with a backward call after. The following AOT_AUTOGRAD_GRAPH gets printed:https://gist.github.com/drisspg/5bbec0d7ca48c961e8e9de7768147245

Offending line: `_scaled_dot_product_flash_attention_backward = torch.ops.aten._scaled_dot_product_flash_attention_backward.default(tangents_1, primals_1, primals_2, primals_3, getitem, getitem_1, getitem_2, getitem_3, 128, 128, 0.0, False, 1, 1)`

The 1s get hardcoded to the backward call. The value is not 1 in reality. They are the default generators PhiloxSeed (this may be constant)
And the PhiloxOffset: this is deff not constant. I would love If there was a way to say this is dynamic but not .. dynamic shapes?

Reply:
I originally tried returning CPU tensors but that didn't really work with compile.  After speaking with @eellison the suggestion was lets try and wrap these up into a cuda kernel. The problem ( as I see it ) is that the kernel would now take in data_ptrs to single element tensors.  The kernel would then read from this ptr.. would this tank perf because of reads?

Reply:
Could I just insert graph breaks somewhere precise?